content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md

```diff
@@ -84,6 +86,47 @@ kubeadm to tell it what to do. This crashloop is expected and normal.
 After you initialize your control-plane, the kubelet runs normally.
 {{< /note >}}
 
+#### Network setup
+
+kubeadm similarly to other Kubernetes components tries to find a usable IP on
+the network interface associated with the default gateway on a host. Such
+an IP is then used for the advertising and/or listening performed by a component.
+
+To find out what this IP is on a Linux host you can use:
+
+```shell
+ip route # Look for the line starting with "default via"
+```
+
+Kubernetes components do not accept custom network interface as an option,
+therefore a custom IP address must be passed as a flag to all components instances
+that need such a custom configuration.
+
+To configure the API server advertise address for control plane nodes created with both
+`init` and `join`, the flag `--apiserver-advertise-address` can be used.
+Preferably, this option can be set in the [kubeadm API](/docs/reference/config-api/kubeadm-config.v1beta3)
+as `InitConfiguration.localAPIEndpoint` and `JoinConfiguration.controlPlane.localAPIEndpoint`.
+
+For kubelets on all nodes, the `--node-ip` option can be passed in kubeadm's
+`InitConfiguration.nodeRegistration.kubeletExtraArgs` or
+`JoinConfiguration.nodeRegistration.kubeletExtraArgs`.
```

Comment:
we already do this in other places in the docs. they are main structures that are passed to the kubeadm init/join workflow.

there is no way to link to individual fields in https://kubernetes.io/docs/reference/config-api/kubeadm-config.v1beta3/

Reply:
I'd call it something like
> `.nodeRegistration.kubeletExtraArgs` inside a kubeadm configuration file
> (InitConfiguration or JoinConfiguration)

and do that even if existing pages use the `InitConfiguration.nodeRegistration.kubeletExtraArgs` style.

Otherwise, I think we'll confuse readers even before they've set up their first cluster, possibly enough to put them off trying it out. It would have confused me as a beginner, for sure.

Reply:
updated.